app/logic/events.py

```diff
@@ -164,8 +162,6 @@ def saveEventToDb(newEventData):
             # Create or update the event
             if isNewEvent:
                 eventRecord = Event.create(**eventData)
-                if 'program' in newEventData:
-                    ProgramEvent.create(program=newEventData['program'], event=eventRecord)
```

Comment:
You removed how we used to set program, but I don't see where you save the new location of program